actionpack/lib/action_controller/metal/request_forgery_protection.rb

```diff
@@ -240,6 +284,22 @@ def handle_unverified_request
       end
     end
 
+    class SessionStore
+      def fetch(request)
+        request.session[:_csrf_token]
+      end
+
+      def store(request, csrf_token)
+        request.session[:_csrf_token] = csrf_token
+      end
+    end
+
+    class NotEmptySessionStore < SessionStore
+      def store(request, csrf_token)
+        super unless request.session.empty?
+      end
+    end
```

Comment:
I'm really tempted to just make this the default to be honest.

Reply:
Me too... but I do worry we might break, or at least change, some edge case behaviour in a way that makes people sad. I have no opinion on whether we should make it a pluggable strategy or just push a single config option into the implementation, but IMO it does seem worth keeping out options open for now -- perhaps make it a new-app default?